### PR TITLE
Upgrade CLI dependencies

### PR DESCRIPTION
**Pull request by moroine**
**Minor deps upgrades:**
- `@octokit/rest`: `16.33.1` ~> `16.43.1`
- `colors`: `1.3.2` ~> `1.4.0`
- `flowgen`: `1.9.0` ~> `1.10.0`
- `glob`: `7.1.3` ~> `7.1.3`
- `prettier`: `1.18.2` ~> `1.19.1`
- `table`: `5.0.2` ~> `5.4.6`

**Major deps upgrades:**
- `fs-extra`: `7.0.0` ~> `8.1.0`
- `got`: `8.3.2` ~> `10.5.7`
- `mkdirp`: `0.5.1` ~> `1.0.3`
- `rimraf`: `2.6.2` ~> `3.0.2`
- `semver`: `5.5.1` ~> `7.1.3`
- `unzipper`: `0.9.3` ~> `0.10.8`
- `which`: `1.3.1` ~> `2.0.2`
- `yargs`: `12.0.2` ~> `15.1.0`

**Removed deps:**
- `@babel/polyfill`: not needed any more thanks to dropping end-of-life Node versions

**Dev-deps upgrades:**
- `@babel/cli`: `7.1.0` ~> `7.8.4`
- `@babel/core`: `7.1.0` ~> `7.8.4`
- `@babel/preset-env`: `7.1.0` ~> `7.8.4`
- `@babel/preset-flow`: `7.1.0` ~> `7.8.4`
- `babel-core`: **removed**
- `babel-eslint`: `10.0.1` ~> `10.0.3`
- `babel-jest`: `23.6.0` ~> `25.1.0`
- `eslint `: `5.6.0` ~> `6.8.0`
- `eslint-plugin-flowtype`: `2.50.3` ~> `4.6.0`
- `eslint-plugin-prettier`: `2.7.0` ~> `3.1.2`
- `flow-bin`: `0.81.0` ~> `0.118.0` 🚀 
- `jest`: `23.6.0` ~> `25.1.0`

**CI:**
- remove `exit 0` to return error code on failed test

**Misc:**
- Fix `@octokit/rest` deprecation message

**Flow Defs:**
- `mkdirp_v1.0.x` will need to be added later in definitions (will do a separated PR)
- `yargs_v15.0.x` has been created by copy/paste `yargs_v12.0.x` this should be alright as API didn't change much but would require a proper PR.

**Breaking Change:**
- minimum NodeJS required version 10 (following https://nodejs.org/en/about/releases/)

### CODE REVIEW

**Comment by AndrewSouthpaw** [comment]
@moroine Amazing! 🎉  Let me check in with folks to make sure the bump to node >= 10 is acceptable.

**Comment by AndrewSouthpaw** [comment]
Also there's a few CLI tests that are still failing, and they can be reproduced locally with running `./travis.sh`

File: cli/src/commands/install.js [hunk 1]
```diff
@@ -58,67 +57,75 @@ export type Args = {
...
+    .usage(`$0 ${name} - ${description}`)
+    .positional('explicitLibDefs', {
+      describe: 'Explicitly specify packages to install',
+      default: [],
+    })
+    .options({
+      flowVersion: {
+        alias: 'f',
+        describe:
+          'The Flow version that fetched libdefs must be compatible ' + 'with',
```
Comment (pascalduez) [inline cli/src/commands/install.js]: I know it was already there but wondering about the concatenation... Can't see any use.

File: cli/src/lib/npm/npmLibDefs.js [hunk 2]
```diff
@@ -70,11 +70,11 @@ async function extractLibDefsFromNpmPkgDir(
   const pkgDirItems = await fs.readdir(pkgDirPath);
 
   if (validating) {
-    const fullPkgName = `${scope === null ? '' : scope + '/'}${pkgName}`;
+    const fullPkgName = `${scope === null ? '' : scope + '/'}${pkgName}popopko`;
```
Comment (pascalduez) [inline cli/src/lib/npm/npmLibDefs.js]: You might want to remove this? :)

File: cli/src/lib/npm/npmLibDefs.js [hunk 3]
```diff
@@ -88,8 +88,9 @@ async function extractLibDefsFromNpmPkgDir(
       })
       .then()
       .catch(error => {
+        console.log(error);
```
Comment (pascalduez) [inline cli/src/lib/npm/npmLibDefs.js]: Remove?

**Comment by pascalduez** [comment]
@moroine Really nice, thanks!

About the node >= 10 bump, it's okay if we release this as a major version I guess.

File: cli/package.json [hunk 4]
```diff
@@ -13,48 +13,49 @@
   "version": "2.6.2",
   "main": "dist/cli.js",
   "bin": "dist/cli.js",
+  "engines": {
+    "node": ">=10.0.0"
+  },
```
Comment (GAntoine) [inline cli/package.json]: Good timing on this change (now that Node 8 is no longer maintained)

File: cli/src/commands/install.js [hunk 5]
```diff
@@ -58,67 +57,75 @@ export type Args = {
...
-      describe: 'Use cache until specified time in milliseconds',
-      type: 'number',
-    },
-  });
+  return yargs
+    .usage(`$0 ${name} - ${description}`)
+    .positional('explicitLibDefs', {
+      describe: 'Explicitly specify packages to install',
+      default: [],
+    })
```
Comment (GAntoine) [inline cli/src/commands/install.js]: If `explicitLibDefs` is a new param, it should be added to https://github.com/flow-typed/flow-typed/wiki/CLI-Commands-and-Flags when this ships.

Comment (moroine) [reply cli/src/commands/install.js]: It's not a new param, in `flow-typed install axios@0.11.x yargs@15.1.x` `explicitLibDefs = ['axios@0.11.x', 'yargs@15.1.x']`

**Comment by moroine** [comment]
> Also there's a few CLI tests that are still failing, and they can be reproduced locally with running `./travis.sh`

@AndrewSouthpaw test failing in NodeJS 12 is due to change in NodeJS 11  when switching to a stable algorithm. I've made the fix to have a deterministic order (ignoring prerelease).

**Comment by AndrewSouthpaw** [comment]
Great work @moroine ! Merging.

**Comment by AndrewSouthpaw** [comment]
@GAntoine You'll probably need to bump the flow-typed version.

**Comment by GAntoine** [comment]
@AndrewSouthpaw On it, I'll get one out tonight.

**Pull request merged**